### PR TITLE
CI: Building CIRCT using an action

### PR DESCRIPTION
**Pull request by sjalander**
This simplifies the HLS test flow.

### CODE REVIEW

File: .github/workflows/hls.yml [hunk 1]
```diff
@@ -71,46 +31,11 @@ jobs:
       with:
         fetch-depth: '0'
 
-    # --------
-    # Install dependencies
-    # --------
-
     - name: "Install LLVM, Clang, MLIR, and Ninja"
       uses: ./.github/actions/InstallLlvmDependencies
```
Comment (phate) [inline .github/workflows/hls.yml:35]: Do these lines still need to be here? Are the dependencies not installed in the BuildCirct composite action?

File: .github/actions/BuildCirct/action.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      id: cache-circt
+      uses: actions/cache@v3
+      with:
+        path: |
+          jlm-eval-suite/circt/local
+        key: ${{ runner.os }}-circt-${{ steps.get-circt-hash.outputs.hash }}
+
+    # Install dependencies if we didn't hit in the cache
+    - name: "Install LLVM, Clang, MLIR, and Ninja"
+      if: steps.cache-circt.outputs.cache-hit != 'true'
```
Comment (sjalander) [inline .github/actions/BuildCirct/action.yml:31]: The installation of dependencies are conditional

File: .github/workflows/hls.yml (hunk 1 above)
Comment (sjalander) [reply .github/workflows/hls.yml:35]: The dependencies are only installed if you miss in the cache for the build CIRCT action.
See comment below.

Comment (phate) [reply .github/workflows/hls.yml:35]: I missed that the first time. Thanks.

**Comment by phate** [comment]
If you update the branch, I will merge it.

**Review by phate — APPROVED** [review]

**Pull request merged**